packages/txservice/src/provider.ts

```diff
@@ -129,55 +124,7 @@ export class ChainRpcProvider {
     };
 
     return this.resultWrapper<providers.TransactionResponse>(async () => {
-      // Queue up the execution of the transaction.
-      const result = await this.queue.add(
-        async (): Promise<{ response: providers.TransactionResponse | Error; success: boolean }> => {
-          try {
-            // NOTE: This call must be serialized within the queue, as it is depenedent on pending transaction count.
-            transaction.nonce = transaction.nonce ?? (await this.getNonce());
-
-            // Send the transaction.
-            const response = await this.signer.sendTransaction(transaction);
-
-            // Check to see if ethers returned null or undefined for the response; if so, handle as error case.
-            if (response == null) {
-              throw new TransactionServiceFailure("Ethers returned a null or undefined transaction response.", {
-                transaction,
-                response,
-              });
-            }
-
-            // We increment the nonce here, as we know the transaction was sent (response is defined).
-            this.incrementNonce();
-            return { response, success: true };
-          } catch (e) {
-            return { response: e, success: false };
-          }
-        },
-      );
-      if (result.success) {
-        return result.response as providers.TransactionResponse;
-      } else {
-        throw result.response;
-      }
-    });
-  }
-
-  /**
-   * Execute a read transaction using the passed in transaction data, which includes
-   * the target contract which we are reading from.
-   * @param tx Minimal transaction data needed to read from chain.
-   * @returns A string of data read from chain.
-   * @throws ChainError.reasons.ContractReadFailure in the event of a failure
-   * to read from chain.
-   */
-  public readTransaction(tx: ReadTransaction): ResultAsync<string, TransactionError> {
-    return this.resultWrapper<string>(async () => {
-      try {
-        return await this.signer.call(tx);
-      } catch (error) {
-        throw new TransactionReadError(TransactionReadError.reasons.ContractReadError, { error });
-      }
+      return await this.signer.sendTransaction(transaction);
```

Comment:
Where does this tie into the buffer, if at all?

Reply:
dispatch handles queuing now. I separated interests regarding queuing. It gave me more space to work lol. we may just wanna integrate dispatch into this class entirely
and maybe just rename this class dispatch, tbh, cause that's what it is